targets/csharp-unityv2/source/Shared/Internal/PlayFabHttp/PlayFabHTTP.cs

```diff
@@ -0,0 +1,337 @@
+#endif
+            }
+        }
+
+        /// <summary>
+        /// Monobehaviour Update method, logs System info data if gathered.
+        /// </summary>
+        public void Update()
+        {
+            _internalHttp.Update();
+            if (!_systemInfoLogged && PlayFabSettings.EnableRealTimeLogging)
+            {
+                _gatherer.EnqueueToLogger(_logMessageQueue);
+                _systemInfoLogged = true;
+            }
+        }
+
+        /// <summary>
+        /// Monobehaviour FixedUpdate, logs messages when enabled.
+        /// </summary>
+        void FixedUpdate()
+        {
+
+            lock (_logMessageLock)
+            {
+                if (_logMessageQueue.Count > 0 && PlayFabSettings.EnableRealTimeLogging)
+                {
+                    if (_logger == null)
+                    {
+                        return;
+                    }
+                    _logger.Open();
+                    while (_logMessageQueue.Count > 0)
+                    {
+                        _logger.Log(_logMessageQueue.Dequeue());
+                    }
+                    _logger.Close();
+                }
+                else if (_logMessageQueue.Count > 0 && _logMessageQueue.Count > PlayFabSettings.LogCapLimit)
+                {
+                    while (_logMessageQueue.Count > PlayFabSettings.LogCapLimit)
+                    {
+                        _logMessageQueue.Dequeue();
+                    }
+                }
+
+            }
+        }
+
+        #region Helpers
+        /// <summary>
+        /// helper class to register the logger.
+        /// </summary>
+        /// <returns></returns>
+        IEnumerator RegisterLogger()
+        {
+            yield return new WaitForEndOfFrame();
+            if (!string.IsNullOrEmpty(PlayFabSettings.LoggerHost))
+            {
+                //_logger = new UdpPaperTrailLogger(PlayFabSettings.LoggerHost, PlayFabSettings.LoggerPort);
+                _logger = PlayFabSettings.Logger ?? new PlayFabLogger(PlayFabSettings.LoggerHost, PlayFabSettings.LoggerPort);
+#if UNITY_5
+                UnityEngine.Application.logMessageReceivedThreaded += HandleLogOutput;
+#else
+                UniytEngine.Application.RegisterLogCallback(HandleLogOutput);
+#endif
+            }
+        }
+
+        /// <summary>
+        /// Handler to enqueue log messages
+        /// </summary>
+        /// <param name="condition"></param>
+        /// <param name="stacktrace"></param>
+        /// <param name="type"></param>
+        private void HandleLogOutput(string condition, string stacktrace, LogType type)
+        {
+            var message = condition;
+            if (type == LogType.Log || type == LogType.Warning)
+            {
+                //logger.Log(string.Format("{0}: {1}", type.ToString(), message));
+                _logMessageQueue.Enqueue(string.Format("{0}: {1}", type.ToString(), message));
```

Comment:
lets discuss at a later time, I feel this does need to be here.. and it currently is not using any sort of multithreading.